### PR TITLE
Update ready conditions for GKENetworkParamSet

### PR DESCRIPTION
**Pull request by synth-beat**
Adding status conditions to be used for validation updates for GNPs with `NetworkAttachment`.

### CODE REVIEW

**Comment by synth-beat** [comment]
@shouri please review

**Comment by synth-beat** [comment]
@mskrocki @aojea

File: apis/network/v1/gkenetworkparamset_types.go
Comment (mskrocki) [inline apis/network/v1/gkenetworkparamset_types.go:140]: Why deprecate and create less specific error?

Comment (synth-beat) [reply apis/network/v1/gkenetworkparamset_types.go:140]: L3 networks with network attachments do not have secondary ranges. When using these errors in validation, an error string with specific information will still be returned.

Comment (mskrocki) [reply apis/network/v1/gkenetworkparamset_types.go:140]: Dont think that matters. You just need to adjust on when this error is set i.e. when there is no NetworkAttachment defined.

Comment (synth-beat) [reply apis/network/v1/gkenetworkparamset_types.go:140]: I'll update these ready conditions to have specific errors rather than "L3GNPInvalid", which will include keeping "L3SecondaryMissing"

File: apis/network/v1/gkenetworkparamset_types.go
Comment (shouri007) [reply apis/network/v1/gkenetworkparamset_types.go:143]: Sorry, just saw @mskrocki also having the same question. Feel free to close this one out.

File: apis/network/v1/gkenetworkparamset_types.go
Comment (synth-beat) [reply apis/network/v1/gkenetworkparamset_types.go]: Updated -- there was actually no need to change the network x GNP conditions except for adding "NetworkAttachmentUnsupported".

**Review by mskrocki — APPROVED** [review]

File: apis/network/v1/gkenetworkparamset_types.go
Comment (shouri007) [inline apis/network/v1/gkenetworkparamset_types.go:132]: When invalid combination of fields is specified, I believe the entire config is valid rather than just an argument. I think `GNPConfigInvalid` better conveys that and we could trim `Argument` from it

Comment (synth-beat) [reply apis/network/v1/gkenetworkparamset_types.go:132]: Makes sense -- updating name

**Review by mskrocki — APPROVED** [review]

**Comment by synth-beat** [comment]
@aojea Can you help get this in? Maciej's approval doesn't seem to count for write access, even though he's in the OWNERS.

**Comment by synth-beat** [comment]
@bowei Are you able to help merge this in?

**Comment by aojea** [comment]
@yswe can you sync with @bowei and @synth-beat to find a model of approval that fits current teams structures?

I prefer to not be in the critical path of possible

**Comment by aojea** [comment]
I think the problem is that this repo uses GitHub approvals, you need to approve in the GitHub code review button 
Screenshot_20240702-194512

**Review by yswe — APPROVED** [review]

**Pull request merged**
